app/reducers/index.js

```diff
@@ -89,7 +89,12 @@ const appReducer = combineReducers(reducers);
 
 export default function rootReducer(state: State, action: Action) {
   if (action.type === User.LOGOUT) {
-    return appReducer(undefined, action);
+    return appReducer(
+      {
+        routing: state.routing
```

Comment:
This will save the router info and history, but that shouldn't matter since we now use `push` instead of just `replace`.